drivers/flash/soc_flash_nrf_mram.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+
+#include <zephyr/drivers/flash.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/barrier.h>
+
+LOG_MODULE_REGISTER(flash_nrf_mram, CONFIG_FLASH_LOG_LEVEL);
+
+#define DT_DRV_COMPAT nordic_mram
+
+#define MRAM_START DT_INST_REG_ADDR(0)
+#define MRAM_SIZE  DT_INST_REG_SIZE(0)
+
+#define MRAM_WORD_SIZE DT_INST_PROP(0, write_block_size)
+#define MRAM_WORD_MASK (MRAM_WORD_SIZE - 1)
+
+#define ERASE_VALUE 0xff
+
+#if defined(CONFIG_SOC_FLASH_NRF_MRAM_ONE_BYTE_WRITE_ACCESS)
+#define WRITE_BLOCK_SIZE 1
+#else
+#define WRITE_BLOCK_SIZE MRAM_WORD_SIZE
+#endif
+
+#define PAGE_SIZE  CONFIG_SOC_FLASH_NRF_MRAM_LAYOUT_PAGE_SIZE
```

Comment:
Can we do that via DTS?

Reply:
The mram dts instance does definitely have an erase and write block size set right? Because if not a lot of things will outright just not work, e.g. mcuboot

Reply:
We have to use pages for now as there is a lot of code that depends on int. Even when we provide ability to work with devices that do not require erase, which is in progress, there are still parts of system that do things by pages, like FCB for example. MCUboot is other example of that, as Jamie has mentioned.

Reply:
For the record, I retested this driver with FCB (as Settings backend in `samples/subsys/settings`) and it works OK, except IIRC it only works with write block size <= 4.

Reply:
Added `erase-block-size` property.